### PR TITLE
fix: add workflowgraph when parsing pipeline template

### PR DESCRIPTION
**Pull request by VonnyJap**
## Context

Workflow graph is not generated when parsing the pipeline template. 

## Objective

Generates the workflow graph when parsing the pipeline template. 

## License

I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -325,18 +333,24 @@ async function parsePipelineTemplate({ yaml }) {
...
+    console.log('newJobs: ', newJobs);
+
     pipelineTemplate.config.jobs = newJobs;
 
-    // Add workflowGraph
-    const { doc } = await phaseValidateFunctionality({
-        flattenedDoc: pipelineTemplate.config,
-        isPipelineTemplate: true
-    });
+    // // Add workflowGraph
```
Comment (tkyi) [inline index.js]: Commented code

File: index.js [hunk 2]
```diff
@@ -325,18 +333,24 @@ async function parsePipelineTemplate({ yaml }) {
  */
 async function validatePipelineTemplate({ yaml, templateFactory }) {
     const pipelineTemplate = await parsePipelineTemplate({ yaml });
+
+    console.log('pipelineTemplate.config: ', pipelineTemplate.config);
```
Comment (tkyi) [inline index.js]: Console logs

**Review by tkyi — APPROVED** [review]

**Comment by tkyi** [comment]
Screenshot 2024-08-11 at 3 49 49 PM
Might be related

**Comment by VonnyJap** [comment]
I suspect that we somehow persist the workflow graph during the pipeline template creation so it failed to comply to this schema https://github.com/screwdriver-cd/data-schema/blob/master/config/base.js#L96-L109, the workaround for now is perhaps to add workflowGraph as one of the allowed keys, but I think by design, workflow graph should be calculated and not persisted.

**Pull request merged**